### ISSUE TITLE
Use isolates to perform decoding

### ISSUE THREAD

**Issue by aarol** [AdityaMulgundkar/wallpaper_manager#25]
Currently the performance is not very good, because the image decoding is done on unefficiently [here](https://github.com/AdityaMulgundkar/wallpaper_manager/blob/153c94139c8b9d9c089567b6cf6a4f02b156741b/android/src/main/java/com/mulgundkar/wallpaper_manager/WallpaperManagerPlugin.java#L126). The decoding happens on the main thread which hangs up the application. The solution to this is to run the decoding process in a separate isolate and sending the bytearray throught the method channel.

I have made a [barebones example](https://github.com/aaroluo/wallpapertest) of this approach which uses very big images and the performance gain is noticeable, there are no frame drops on my emulator.

I'm hoping someone with more knowledge of the package could implement this.

**Comment by edwinmacalopu** [comment]
Yes, this implementation is necessary for very large images and does not block the application

**Comment by Praveen-0603** [comment]
@aarol can you pls write the code for settingwallpaper from file too?

**Comment by Praveen-0603** [comment]
Yes, I created a package, thanks @aarol for the idea,
Search Wallpaper_manager_flutter in pub.dev

**Comment by MabudAlam** [comment]
My application is lagging and frames are dropping constantly. How can I resolve this?

**Comment by aarol** [comment]
> My application is lagging and frames are dropping constantly. How can I resolve this?

Are you using the code from the example? It should lag only for a small period probably due to memory being copied from the isolate.

**Comment by PraveenGongada** [comment]
> My application is lagging and frames are dropping constantly. How can I resolve this?

https://pub.dev/packages/wallpaper_manager_flutter use this package 

Use this package, there will be no lag even for 4k images, thanks to @aarol  for the idea.

**Comment by MabudAlam** [comment]
Thanks @Praveen-0603  this  [package](https://pub.dev/packages/wallpaper_manager_flutte) solved all my problems.
